python/helpers.py

```diff
@@ -0,0 +1,268 @@
+import requests
+import base64
+import math
+from dotenv import load_dotenv
+from borsh_construct import CStruct, U8, U32, U64
+from PIL import Image
+import easyocr
+import io
+import boto3
+import re
+import time
+import imageio.v3 as iio
+
+load_dotenv()
+dynamodb = boto3.resource('dynamodb')
+imageTable = dynamodb.Table('imageTable')
+treeTable = dynamodb.Table('treeTable')
+
+reader = easyocr.Reader(['en'])
+
+KEYWORDS = [
+    "contains_emoji",
+    "proof_lengthImpossible",
+    "imagecontains_url",
+    "not_contains_emoji",
+    "not_proof_lengthImpossible",
+    "not_imagecontains_url",
+]
+
+
+EMOJI_PATTERN = re.compile(
+    "["
+    "\U0001F1E0-\U0001F1FF"  # flags (iOS)
+    "\U0001F300-\U0001F5FF"  # symbols & pictographs
+    "\U0001F600-\U0001F64F"  # emoticons
+    "\U0001F680-\U0001F6FF"  # transport & map symbols
+    "\U0001F700-\U0001F77F"  # alchemical symbols
+    "\U0001F780-\U0001F7FF"  # Geometric Shapes Extended
+    "\U0001F800-\U0001F8FF"  # Supplemental Arrows-C
+    "\U0001F900-\U0001F9FF"  # Supplemental Symbols and Pictographs
+    "\U0001FA00-\U0001FA6F"  # Chess Symbols
+    "\U0001FA70-\U0001FAFF"  # Symbols and Pictographs Extended-A
+    "\U00002702-\U000027B0"  # Dingbats
+    "\U000024C2-\U0001F251" 
+    "]+",
+    flags = re.UNICODE
+)
+
+HEADER_SCHEMA = CStruct(
+  "versionedHeader" / U8,
+  "notSure" / U8,
+  "maxBufferSize" / U32,
+  "maxDepth" / U32,
+  "authority" / U8[32],
+  "creationSlot" / U64,
+  "padding" / U8[6],
+  "sequenceNumber" / U64,
+  "activeIndex" / U64,
+  "bufferSize" / U64,
+)
+
+def get_image_words(image_url):
+    """
+    Get words from a given image url using OCR 
+    If an mp4 file is given, get the image from the first frame
+    image_url: the url of the image or video to get words from
+    """
+    response = imageTable.get_item(
+        Key={
+            'url': image_url,
+        }
+    )
+    if "Item" in response:
+        if "words" in response["Item"].keys():
+            return response["Item"]["words"]
+        # A malformed entry has made its way into the table
```

Comment:
attempt at removing malformed entries since this is kind of a required field

Reply:
do we have malformed entries rn?

Reply:
as part of running the train script I think I created them for a little bit but we shouldnt now that Ive run them again

Reply:
makes sense, we should always be pushing to `words` even if it's an empty array i believe so this should be redundant but doesn't hurt